src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -114,9 +144,25 @@ trait AttributeComponent {
 
   protected object submissionAttributeQuery extends AttributeQuery[Long, SubmissionAttributeRecord, SubmissionAttributeTable](new SubmissionAttributeTable(_), SubmissionAttributeRecord)
 
-  protected object entityAttributeTempQuery extends AttributeQuery[Long, EntityAttributeRecord, EntityAttributeTempTable](new EntityAttributeTempTable(_), EntityAttributeRecord)
+  protected object entityAttributeTempQuery extends TableQuery(new EntityAttributeTempTable(_)) {
+    def batchInsertAttributes(attributes: Seq[EntityAttributeRecord], transactionId: String) = {
+      insertInBatches(this, attributes.map(tempRecord(_, transactionId)))
+    }
 
-  protected object workspaceAttributeTempQuery extends AttributeQuery[UUID, WorkspaceAttributeRecord, WorkspaceAttributeTempTable](new WorkspaceAttributeTempTable(_), WorkspaceAttributeRecord)
+    def tempRecord(rec: EntityAttributeRecord, transactionId: String) = {
+      EntityAttributeTempRecord(rec.id, rec.entityId, rec.name, rec.valueString, rec.valueNumber, rec.valueBoolean, rec.valueEntityRef, rec.listIndex, rec.listLength, transactionId)
+    }
+  }
+
+  protected object workspaceAttributeTempQuery extends TableQuery(new WorkspaceAttributeTempTable(_)) {
+    def batchInsertAttributes(attributes: Seq[WorkspaceAttributeRecord], transactionId: String) = {
+      insertInBatches(this, attributes.map(tempRecord(_, transactionId)))
+    }
+
+    def tempRecord(rec: WorkspaceAttributeRecord, transactionId: String) = {
+      WorkspaceAttributeTempRecord(rec.id, rec.workspaceId, rec.name, rec.valueString, rec.valueNumber, rec.valueBoolean, rec.valueEntityRef, rec.listIndex, rec.listLength, transactionId)
+    }
+  }
```

Comment:
I really want to make these two - entityAttributeTempQuery and workspaceAttributeTempQuery - generic; the only difference is the Record classes in play. But I'm not sure if it's realistically possible, without bringing up an IDE :)